src/sentry/profiles/task.py

```diff
@@ -221,18 +221,26 @@ def _process_symbolicator_results_for_sample(profile: Profile, stacktraces: List
             frame.pop("context_line", None)
             frame.pop("post_context", None)
 
-        def truncate_stack_needed(frame: dict[str, Any]) -> bool:
-            return bool(frame.get("function", "") == "perf_signal_handler")
+        def truncate_stack_needed(
+            frame: dict[str, Any], stack: List[Any]
+        ) -> Tuple[bool, List[Any]]:
+            # remove top frames related to the profiler
+            return bool(frame.get("function", "") == "perf_signal_handler"), stack[2:]
 
     elif profile["platform"] == "cocoa":
 
-        def truncate_stack_needed(frame: dict[str, Any]) -> bool:
-            return bool(frame.get("instruction_addr", "") == "0xffffffffc")
+        def truncate_stack_needed(
+            frame: dict[str, Any], stack: List[Any]
+        ) -> Tuple[bool, List[Any]]:
+            # remove bottom frames we can't symbolicate
+            return bool(frame.get("instruction_addr", "") == "0xffffffffc"), stack[:-2]
 
     else:
 
-        def truncate_stack_needed(frame: dict[str, Any]) -> bool:
-            return False
+        def truncate_stack_needed(
+            frame: dict[str, Any], stack: List[Any]
+        ) -> Tuple[bool, List[Any]]:
+            return False, []
```

Comment:
you could change the signature of this function to return just the new stack so this becomes

```python
def process_stack(frame: Dict[str, Any], stack: List[Any]):
  return stack
```

And the other 2 can do something similar instead of returning an unused empty list